src/Identity/v3/Service.php

```diff
@@ -31,7 +31,11 @@ public function authenticate(array $options): array
     {
         $authOptions = array_intersect_key($options, $this->api->postTokens()['params']);
 
-        $token = $this->generateToken($authOptions);
+        if (!empty($options['cachedCredential'])) {
+            $token = $this->generateTokenFromCache($options['cachedCredential']);
```

Comment:
imagine that I'm passing in an expired array here, should a fatal error be raised? My initial thought is yes. WDYT?

Reply:
Good point. Validation added.